### ISSUE TITLE
Unable to add nuget package to .Net 6.0 library in VS2022

### ISSUE THREAD

**Issue by WilfBanger** [dotnet/SqlClient#1514]
### Bug
In a brand new empty library targeted at net6.0, I am unable to add the Microsoft.Data.SqlClient package. Nuget reports the below error.

`NU1100: Unable to resolve 'Microsoft.Data.SqlClient.SNI.runtime (>= 2.1.1)' for 'net6.0'.`

I have tried in VS2022 using both the package manager UI and the console with the same result.

### Further technical details
Microsoft.Data.SqlClient version: 6.0.2
.NET target: net6.0
SQL Server version: N/A
Operating system: Windows 10

**Comment by WilfBanger** [comment]
Update: Any version above 2.0.1 results in an error.

**Comment by WilfBanger** [comment]
@JRahnama  You are correct, I do have an offline cache of packages installed so far, by redirecting the Nuget  `globalPackagesFolder` to a shareable local folder. In there is a version of `Microsoft.Data.SqlClient.SNI.runtime` but it is version 2.0.1 and so I would expect that to be ignored due to the higher requirement.

If I try to install `Microsoft.Data.SqlClient.SNI.runtime` directly at 4.0.0 I get the same error result.

`NU1100: Unable to resolve 'Microsoft.Data.SqlClient.SNI.runtime (>= 4.0.0)' for 'net6.0'.'`

Here is my local Nuget.config:
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear/>
    <add key="Packages - Core" value="C:\_dev.environment\.Net\packages.Core" />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
    <add key="Microsoft Visual Studio Offline Packages" value="C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\" />
  </packageSources>
  <packageRestore>
    <add key="enabled" value="True" />
    <add key="automatic" value="True" />
  </packageRestore>
  <bindingRedirects>
    <add key="skip" value="False" />
  </bindingRedirects>
  <config>
    <add key="globalPackagesFolder" value="c:\_dev.environment\.Net\packages.Core" />
  </config>
  <packageManagement>
    <add key="format" value="1" />
    <add key="disabled" value="True" />
  </packageManagement>
</configuration>
```

**Comment by JRahnama** [comment]
@WilfBanger are you using CLI or just VS 2022 IDE?

**Comment by WilfBanger** [comment]
@JRahnama Right now just the built-in inside the VS2022 IDE, which according to the Help->About is NuGet Package Manager   6.0.1

**Comment by WilfBanger** [comment]
@JRahnama From the image you can see that the only sources available are Nuget.org and the Offline MS packages, and that when I attempt to install the package it gives the error.

![image](https://user-images.githubusercontent.com/99358769/153523206-a00f5a25-046f-403e-9be7-c2cdcdd23aad.png)

**Comment by WilfBanger** [comment]
@JRahnama  Ok, finally figured this out. The problem was in the configuration of the base Nuget.config file and nothing to do with the local one I have or the packages. An incomplete new `packageSourceMapping` section had been added where it was missing a default mapping, so, even though the `nuget.org` source had been declared and was showing up in the IDE, it was not being considered a source for downloads.

Adding a
````
<packageSource key="nuget.org"><package pattern="*" /></packageSource>
````
to the `packageSourceMapping` section allows any package not already in the cache to be downloaded from `nuget.org`. It was working for other packages I had tried as they were already in the cache and only this package and it's graph needed downloading.

According to the [packageSourceMapping documentation](https://docs.microsoft.com/en-us/nuget/consume-packages/package-source-mapping) there is no CLI or IDE support for surfacing the settings for this section as yet. The [nuget.config reference page](https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#package-source-mapping-section) that talks about the section doesn't go into as much detail and wasn't clear to me when I read it during my investigations last week that all Ids needed to be covered.
